packages/integrations/mdx/src/rehype-apply-frontmatter-export.ts

```diff
@@ -0,0 +1,49 @@
+import { InvalidAstroDataError } from '@astrojs/markdown-remark';
+import { safelyGetAstroData } from '@astrojs/markdown-remark/dist/internal.js';
+import type { VFile } from 'vfile';
+import { jsToTreeNode } from './utils.js';
+
+export function rehypeApplyFrontmatterExport() {
+	return function (tree: any, vfile: VFile) {
+		const astroData = safelyGetAstroData(vfile.data);
+		if (astroData instanceof InvalidAstroDataError)
+			throw new Error(
+				// Copied from Astro core `errors-data`
+				// TODO: find way to import error data from core
```

Comment:
Now we have `AstroError`:

```js
import { AstroError } from "astro/errors";
```

Reply:
I mostly copied the existing code over. We can follow-up a PR for this improvement 😬